### PR TITLE
Bounding box nested fixes

### PR DESCRIPTION
**Pull request by TrevorDev**
https://github.com/BabylonJS/Babylon.js/issues/4623

### CODE REVIEW

File: src/Cameras/VR/babylon.vrExperienceHelper.ts [hunk 1]
```diff
@@ -921,7 +921,7 @@ module BABYLON {
                 }
 
                 this.raySelectionPredicate = (mesh) => {
-                    return mesh.isVisible;
+                    return mesh.isVisible && mesh.isPickable;
```
Comment (TrevorDev) [inline src/Cameras/VR/babylon.vrExperienceHelper.ts]: @davrous is there a reason this was not in this predicate by default? Would this be okay to add now?

Comment (deltakosh) [reply src/Cameras/VR/babylon.vrExperienceHelper.ts]: Agree this should be like that (with the isPickable)

File: src/Gizmos/babylon.boundingBoxGizmo.ts [hunk 2]
```diff
@@ -329,5 +376,48 @@ module BABYLON {
...
+            }
+            var oldPos = mesh.position.clone()
+            var oldRot = mesh.rotationQuaternion.clone();
+            mesh.rotationQuaternion.set(0,0,0,1);
+            mesh.position.set(0,0,0)
+
+            // Update bounding dimensions/positions   
+            var box = BABYLON.MeshBuilder.CreateBox("box", {size: 1}, mesh.getScene());
+            var boundingMinMax = mesh.getHierarchyBoundingVectors();
+            box.scaling.x = boundingMinMax.max.x-boundingMinMax.min.x;
```
Comment (RaananW) [inline src/Gizmos/babylon.boundingBoxGizmo.ts]: out of curiosity - why not use the subtract (to reference) function?

File: src/Cameras/VR/babylon.vrExperienceHelper.ts (hunk 1 above)
Comment (davrous) [reply src/Cameras/VR/babylon.vrExperienceHelper.ts]: Indeed, this is better like that. Thanks!

File: src/Gizmos/babylon.boundingBoxGizmo.ts (hunk 2 above)
Comment (TrevorDev) [reply src/Gizmos/babylon.boundingBoxGizmo.ts]: I think when I was writing it, I had more complex logic here, I just updated this to use subtract here.

**Pull request merged**